cmd/juju/caas/add.go

```diff
@@ -363,10 +388,45 @@ var noRecommendedStorageErrMsg = `
 
 // Run is defined on the Command interface.
 func (c *AddCAASCommand) Run(ctx *cmd.Context) (err error) {
+	if c.hostCloudRegion != "" {
+		ctx.Infof("region flag is DEPRECATED. Use 'add-k8s name cloud/region' instead")
+	}
+	if c.cloud != "" {
+		ctx.Infof("cloud flag is DEPRECATED. Use 'add-k8s name cloud/region' instead")
+	}
 	if err := c.verifyName(c.caasName); err != nil {
 		return errors.Trace(err)
 	}
 
+	// validate provided cloud/region
+	cloudClient, err := c.cloudAPIFunc()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	defer cloudClient.Close()
+
+	cloudRegion := ""
+	if c.cloudRegion != "" {
+		cloudRegionParams := common.CloudRegionValidationParams{
+			LocalStore:        c.Store,
+			RemoteCloudClient: cloudClient,
+			LocalOnly:         c.Local,
+			CloudRegion:       c.cloudRegion,
+			Command:           c.Info().Name,
+			UseDefaultRegion:  true,
+		}
+		var cloud jujucloud.Cloud
+		var err error
+		_, cloud, cloudRegion, err = common.ParseCloudRegionMaybeDefaultRegion(cloudRegionParams)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		// TODO (anastasiamac 2019-04-24) For some reason everywhere after this point,
+		// this changes from <cloud name>/<region> to <cloud type>/<region>..
```

Comment:
Technically it is supposed to be cloud type.
But we allow users to specify cloud name as well and that is translated to cloud type